src/components/file/index.tsx

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import React, { ReactElement, useState, useEffect, useContext } from "react";
+import { Url, UrlString, saveFileInContainer } from "@inrupt/solid-client";
+import { SessionContext } from "../../context/sessionContext";
+import { CommonProperties } from "../../helpers";
+
+export type Props = {
+  saveLocation?: Url | UrlString;
+  inputProps?: React.InputHTMLAttributes<HTMLInputElement>;
+} & CommonProperties;
+
+export function FileUpload({
+  saveLocation,
+  onError,
+  onSave,
+  inputProps,
+  ...other
+}: Props & React.HTMLAttributes<HTMLSpanElement>): ReactElement {
+  const { fetch } = useContext(SessionContext);
+  let error: Error | undefined;
+
+  useEffect(() => {
+    if (error && onError) {
+      onError(error);
+    }
+  }, [error, onError]);
+
+  const [, setErrorState] = useState<string | null>();
+
+  const handleChange = async (e: React.FormEvent<HTMLInputElement>) => {
+    const target = e.target as HTMLInputElement;
+    if (target.files) {
+      try {
+        if (saveLocation) {
+          if (target.files[0].name) {
+            const savedFile = await saveFileInContainer(
+              saveLocation,
+              target.files[0],
+              {
+                slug: target.files[0].name,
+                fetch,
+              }
+            );
+          } else {
+            const savedFile = await saveFileInContainer(
+              saveLocation,
+              target.files[0],
+              {
+                fetch,
+              }
+            );
+          }
+        } else {
+          setErrorState(() => {
```

Comment:
No need to throw an error here - it's a required property anyways. It will fail to compile on typescript and error at runtime in React.